cmd/cmd_object.go

```diff
@@ -224,71 +232,153 @@ $ gnfd-cmd object mirror --destChainId 97 --bucketName yourBucketName --objectNa
-			return fmt.Errorf("upload file larger than 10G ")
+
+		filePathList := make([]string, 0)
+		argNum := ctx.Args().Len()
+		for i := 0; i < argNum-1; i++ {
+			filePathList = append(filePathList, ctx.Args().Get(i))
 		}
 
-		// Open the referenced file.
-		file, err := os.Open(filePath)
-		if err != nil {
-			return err
+		var needUploadMutiFiles bool
+		if len(filePathList) > 1 {
+			needUploadMutiFiles = true
 		}
-		defer file.Close()
-		fileReader = file
 
-		urlInfo = ctx.Args().Get(1)
-		bucketName, objectName, err = getObjAndBucketNames(urlInfo)
-		if err != nil {
+		// upload multiple files
+		if needUploadMutiFiles {
+			urlInfo = ctx.Args().Get(argNum - 1)
 			bucketName = ParseBucket(urlInfo)
 			if bucketName == "" {
 				return toCmdErr(errors.New("fail to parse bucket name"))
 			}
-			// if the object name has not been set, set the file name as object name
-			objectName = filepath.Base(filePath)
+
+			for idx, fileName := range filePathList {
+				nameList := strings.Split(fileName, "/")
+				objectName = nameList[len(nameList)-1]
+				objectSize, err = parseFileByArg(ctx, idx)
+				if err != nil {
+					return toCmdErr(err)
+				}
+
+				if err = uploadFile(bucketName, objectName, fileName, urlInfo, ctx, gnfdClient, false, false, objectSize); err != nil {
+					fmt.Println("upload object:", objectName, "err", err)
+				}
+			}
+		} else {
+			// upload single file
+			objectSize, err = parseFileByArg(ctx, 0)
+			if err != nil {
+				return toCmdErr(err)
+			}
+			urlInfo = ctx.Args().Get(1)
+			bucketName, objectName, err = getObjAndBucketNames(urlInfo)
+			if err != nil {
+				bucketName = ParseBucket(urlInfo)
+				if bucketName == "" {
+					return toCmdErr(errors.New("fail to parse bucket name"))
+				}
+				// if the object name has not been set, set the file name as object name
+				objectName = filepath.Base(filePathList[0])
+			}
+			if err = uploadFile(bucketName, objectName, filePathList[0], urlInfo, ctx, gnfdClient, false, true, objectSize); err != nil {
+				return toCmdErr(err)
+			}
 		}
 	}
 
-	gnfdClient, err := NewClient(ctx, false)
+	return nil
+}
+
+// uploadFolder upload folder and the files inside to bucket in a recursive way
+func uploadFolder(urlInfo string, ctx *cli.Context,
+	gnfdClient client.Client) error {
+	// upload folder with recursive flag
+	bucketName := ParseBucket(urlInfo)
+	if bucketName == "" {
+		return errors.New("fail to parse bucket name")
+	}
+
+	folderName := ctx.Args().Get(0)
+	fileInfo, err := os.Stat(folderName)
 	if err != nil {
 		return err
 	}
 
-	c, cancelCreateBucket := context.WithCancel(globalContext)
-	defer cancelCreateBucket()
+	if !fileInfo.IsDir() {
+		return errors.New("failed to parse folder path with recursive flag")
+	}
+	fileInfos := make([]os.FileInfo, 0)
+	filePaths := make([]string, 0)
+	listFolderErr := filepath.Walk(folderName, func(path string, info os.FileInfo, err error) error {
+		if !info.IsDir() {
+			fileInfos = append(fileInfos, info)
```

Comment:
seems fileInfo is not being used below

Reply:
used the info.size as  the objectSize . ` uploadFile(.., info.Size())`